pydantic_core/core_schema.py

```diff
@@ -1867,6 +1922,11 @@ def __call__(
         ...
 
 
+class NoInfoWrapValidatorFunctionSchema(TypedDict):
+    type: Literal['no-info']
+    function: NoInfoWrapValidatorFunction
+
+
 class GeneralWrapValidatorFunctionSchema(TypedDict):
     type: Literal['general']
     function: GeneralWrapValidatorFunction
```

Comment:
Why are we keeping multiple types here instead of adding an `info_field` boolean like with serializers?

Reply:
because the difference between a "field" and "general" validator is just in the type passed to the `info` parameter.

Hence when we remove `info`, there's no difference between them, thus we only need three options:
* no-info
* field
* general

Not 4.